Sources/SwiftTreeSitter/Node.swift

```diff
@@ -82,16 +82,55 @@ extension Node {
         return TSRange(points: pointRange, bytes: byteRange)
     }
 
+    public var isNull: Bool {
+        return ts_node_is_null(internalNode)
+    }
+
+    public var isExtra: Bool {
+        return ts_node_is_extra(internalNode)
+    }
+
+    public var isNamed: Bool {
+        return ts_node_is_named(internalNode)
+    }
+
+    public var isMissing: Bool {
+        return ts_node_is_missing(internalNode)
+    }
+
+    public var hasChanges: Bool {
+        return ts_node_has_changes(internalNode)
+    }
+
+    public var hasError: Bool {
+        return ts_node_has_error(internalNode)
+    }
+
+    public func childByFieldName(fieldName: String) -> Node? {
```

Comment:
Yes. The one you suggested is better. However, if you look at the call site it will look something like `child(by: "func_name")`. It does not give enough clarity at the call site only because the parameter type is not specific. User can pass any string here.

The signature that you have suggested would look more appropriate if the method had a context-specific parameter type like `child(by: fieldName FieldName` then the call site would look something like `child(by: .funcName)`. As the parameter is very specific here, the chances of user making mistakes will be less

Hence, I would like to propose `childBy(fieldName: String)` which serves as a good middle ground.

Let me know your thoughts 😊

Reply:
Yeah, good thinking! But the trailing preposition on the method name just looks a little non-standard. What do you think about `child(byFieldName: String` ? This also matches the existing `child(at index: Int)` better.

Now that I'm looking at it more close, I really think child *with* field name sounds more natural. But, I'd prefer to stick closely to the C API.

Reply:
Yes. Let's go ahead with `child(byFieldName: String`. I have updated the PR